components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3317,20 +3322,32 @@ public String getOpenAPIDefinition(String apiId, String organization) throws API
 
     @Override
     public String invokeApiChatExecute(String apiChatRequestId, String requestPayload) throws APIManagementException {
-        return APIUtil.invokeAIService(APIConstants.AI.API_CHAT_ENDPOINT,
-                APIConstants.AI.API_CHAT_AUTH_TOKEN, APIConstants.AI.API_CHAT_EXECUTE_RESOURCE, requestPayload,
+        ApiChatConfigurationDTO configDto = ServiceReferenceHolder.
+                getInstance().getAPIManagerConfigurationService().getAPIManagerConfiguration().getApiChatConfigurationDto();
+        return APIUtil.invokeAIService(configDto.getEndpoint(),
+                configDto.getAccessToken(), configDto.getExecuteResource(), requestPayload,
                 apiChatRequestId);
     }
 
     @Override
     public String invokeApiChatPrepare(String apiId, String apiChatRequestId, String organization)
             throws APIManagementException {
-        String swaggerDefinition = getOpenAPIDefinition(apiId, organization);
-        String payload = "{\"openapi\": " + swaggerDefinition + "}";
-        String prepareResponse = APIUtil.invokeAIService(APIConstants.AI.API_CHAT_ENDPOINT,
-                APIConstants.AI.API_CHAT_AUTH_TOKEN, APIConstants.AI.API_CHAT_PREPARE_RESOURCE, payload,
-                apiChatRequestId);
-        return prepareResponse;
+        try {
+            // Generate the payload for prepare call
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode openAPIDefinitionJsonNode = objectMapper.readTree(getOpenAPIDefinition(apiId, organization));
+            ObjectNode payload = objectMapper.createObjectNode();
+            payload.set("openapi", openAPIDefinitionJsonNode);
+
+            ApiChatConfigurationDTO configDto = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                    .getAPIManagerConfiguration().getApiChatConfigurationDto();
+            return APIUtil.invokeAIService(configDto.getEndpoint(), configDto.getAccessToken(),
+                    configDto.getPrepareResource(), payload.toString(), apiChatRequestId);
+        } catch (JsonProcessingException e) {
+            String error = "Error while parsing OpenAPI definition to JSON";
```

Comment:
Let's include some sort of an identifier to identify the API

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12385/commits/e23b99dea51b29c978fed6b6fefb746969f086e8